bin/all_sky_search/pycbc_template_kde_calc

```diff
@@ -52,44 +59,50 @@ def optimizedparam(sampleval, nfold=2):
     maxFOM = FOM[(optbw, optalpha)]
     return optbw, optalpha
 
+assert len(args.fit_param) == len(args.log_param)
+
 if args.make_signal_kde and args.make_template_kde:
-    raise Exception("Choose only one option out of --make-signal-kde and --make-template-kde")
+    parser.error("Choose only one option out of --make-signal-kde and --make-template-kde")
+
+#calling template data and constructing template_kde_hdf_file over template params
+temp_file = h5py.File(args.template_file, 'r')
+mass1 = temp_file['mass1']
+tid = numpy.arange(len(mass1))  #tid : integer or array of int for Indices of the entries to be returned
+mass1, mass2, spin1z, spin2z = triggers.get_mass_spin(temp_file, tid)
 
-#calling template data and constructing template_kde_hdf file over template params
-template_file = h5py.File(args.template_file, 'r')
-mass1_n = template_file['mass1'][:]
-mass2_n = template_file['mass2'][:]
-spin1z_n = template_file['spin1z'][:]
-spin2z_n = template_file['spin2z'][:]
-chi_eff_n = convert.chi_eff(mass1_n, mass2_n, spin1z_n, spin2z_n)
-mchirp_n = convert.mchirp_from_mass1_mass2(mass1_n, mass2_n)
-eta_n = convert.eta_from_mass1_mass2(mass1_n, mass2_n)
+template_pars = []
+for param, slog in zip(args.fit_param, args.log_param):
+    data = triggers.get_param(param, args, mass1, mass2, spin1z, spin2z)
+    if slog in ['False']:
+        logging.info('Using param: %s', param)
+        template_pars.append(data)
+    elif slog in ['True']:
+        logging.info('Using log param: %s', param)
+        template_pars.append(numpy.log(data))
+    else:
+        raise ValueError("invalid log param argument, use 'True', or 'False'")
+sample1 = numpy.vstack((template_pars)).T
```

Comment:
can this be called something more informative than 'sample1' ? eg 'temp_samples' to make it clear that it concerns the template parameters?